### PR TITLE
[docs] Improve English and fix 404 link

### LINKED ISSUE #1814: Broken link to KeyboardDateTimePicker on v4 page
At the bottom, https://material-ui-pickers.mui-org.now.sh/demo/datetime-picker#components-api links to https://material-ui-pickers.mui-org.now.sh/api/KeyboardDateTimePicker.

Looks like KeyboardDateTimePicker was removed, but MobileDateTimePicker is not included in the sidebar.

image

### PR DESCRIPTION
**Pull request by dandv**
This PR hopefully closes #1814 partially.

### CODE REVIEW

File: docs/pages/demo/datetime-picker/index.mdx [hunk 1]
```diff
@@ -32,9 +32,10 @@ DateTimePicker is a complex control, so you may consider using `MobileDateTimePi
 
 #### Date & Time validation
 
-It is possible to restrict date and time selection in couple of different ways. If you need to restrict time selection
-you could use `minDateTime` to restrict selecting date and time before particular moment of time.
-Or you can use `minTime` to disable a part of time in each day.
+It is possible to restrict date and time selection in two ways:
+
+* using `minDateTime`, you can restrict time selection before a particular monet in time
```
Comment (dmtrKovalenko) [inline docs/pages/demo/datetime-picker/index.mdx]:
```suggestion
* using `minDateTime`, you can restrict time selection before a particular moment in time
```

Comment (dandv) [reply docs/pages/demo/datetime-picker/index.mdx]: Why doesn't GitHub have spell checking in their CodeMirror instance :-/

File: docs/pages/demo/datetime-picker/index.mdx [hunk 2]
```diff
@@ -32,9 +32,10 @@ DateTimePicker is a complex control, so you may consider using `MobileDateTimePi
 
 #### Date & Time validation
 
-It is possible to restrict date and time selection in couple of different ways. If you need to restrict time selection
-you could use `minDateTime` to restrict selecting date and time before particular moment of time.
-Or you can use `minTime` to disable a part of time in each day.
+It is possible to restrict date and time selection in two ways:
+
+* using `minDateTime`, you can restrict time selection before a particular moment in time
+* using `minTime`, you can disable selecting times before a certain time each day
```
Comment (oliviertassinari) [inline docs/pages/demo/datetime-picker/index.mdx]: To match the markdown style of the main repository

```suggestion
- using `minDateTime`, you can restrict time selection before a particular moment in time
- using `minTime`, you can disable selecting times before a certain time each day
```

File: docs/pages/demo/datetime-picker/index.mdx [hunk 3]
```diff
@@ -46,4 +47,4 @@ Here are some examples of heavily customized date & time pickers
 
 #### Components API
 
-<LinkedComponents components={['DateTimePicker', 'KeyboardDateTimePicker']} />
+<LinkedComponents components={['DateTimePicker', 'MobileDateTimePicker']} />
```
Comment (oliviertassinari) [inline docs/pages/demo/datetime-picker/index.mdx]: 3 other pages of the documentation are impacted. I think that whatever solution we go with, it should impact them too.

**Comment by dmtrKovalenko** [comment]
I think we should mention not only ‘minTime’ and `minDateTime`  but also `max` analogues.

File: docs/pages/demo/datetime-picker/index.mdx (hunk 2 above)
Comment (dmtrKovalenko) [inline docs/pages/demo/datetime-picker/index.mdx]:
```suggestion
- using `minDateTime`/`maxDateTime`, you can restrict time selection before or after a particular moment in time
- using `minTime`/`maxTime`, you can disable selecting times before a certain time each day
```

File: docs/pages/demo/datetime-picker/index.mdx [hunk 4]
```diff
@@ -32,9 +32,10 @@ DateTimePicker is a complex control, so you may consider using `MobileDateTimePi
 
 #### Date & Time validation
 
-It is possible to restrict date and time selection in couple of different ways. If you need to restrict time selection
-you could use `minDateTime` to restrict selecting date and time before particular moment of time.
-Or you can use `minTime` to disable a part of time in each day.
+It is possible to restrict date and time selection in two ways:
+
+- using `minDateTime`/`maxDateTime`, you can restrict time selection before or after a particular moment in time
+- using `minTime`/`maxTime`, you can disable selecting times before a certain time each day
```
Comment (dmtrKovalenko) [inline docs/pages/demo/datetime-picker/index.mdx]:
```suggestion
- using `minTime`/`maxTime`, you can disable selecting times before or after a certain time each day respectively
```

**Review by oliviertassinari — APPROVED** [review]
I have updated the markdown to fix the 404 links as well as uniformize the generated markdown between the components. A couple of thoughts:

- I wonder what should be our strategy to document the Mobile, Desktop, and Static variants of the components? I wouldn't be against a standalone component file, e.g. `MobileDatePicker.js` and API page.
- The heading structure of the docs is wrong. The h1 > h2 > h3 structure should be continuous (for a11y). For instance, there is no h1, it starts (❌) from h2, then there is no h3, to jumps (❌) to h4. This gave me a new item idea to list into https://github.com/mui-org/material-ui/issues/20960: the accessibility of the docs. A report by the Chrome Wave extension:
wave

- There are a couple of UI jumps between the SSR rendered and hydrated versions of the components, I'm surprised nobody reported it yet (I guess people use it for client-side only dashboard, not e-commerce much). I think that it will come with more usage, happy to be lazy and wait.

**Comment by dmtrKovalenko** [comment]
Answering:
1) No, we do not need to link API for each component, because they are mostly identical. In v3 `KeyboardDatePicker` and `DatePicker` had different props
2) Totally right, shame on me 👎 
3) It is because an idea to prerender time on the server is wrong. Because of timezones conflict, it can be useful for some edge-cases like a google calendar app or some meeting app. But I think just nobody using SSR with picker content (like calendar or clock)

**Comment by oliviertassinari** [comment]
> But I think just nobody using SSR with picker content (like calendar or clock)

Yeah, I guess it's a much smaller segment of the market 😆.

**Review by dmtrKovalenko — APPROVED** [review]

**Pull request merged**